packages/flutter_tools/lib/src/commands/build_aot.dart

```diff
@@ -100,8 +100,12 @@ String buildAotSnapshot(
   } else {
     String artifactsDir = tools.getEngineArtifactsDirectory(platform, buildMode).path;
     entryPointsDir = artifactsDir;
-    String hostToolsDir = path.join(artifactsDir, getNameForHostPlatform(getCurrentHostPlatform()));
-    genSnapshot = path.join(hostToolsDir, 'gen_snapshot');
+    if (platform == TargetPlatform.ios) {
```

Comment:
btw the dream is to have all this platform-specific logic abstracted out into virtual methods on the platform objects so you just invoke the right method on the platform object and the right thing happens. Ideally we'd never check the platform type in non-platform-specific code.